### PR TITLE
chore: Switch to smaller cache dep

### PR DESCRIPTION
**Pull request by mkmik**
Another attempt at trimming our deps, like #1747 but this crate supports TTLs too (not used here).

This crate also doesn't suffer from the thundering herd issue, and it waits until the first thread that misses a key loads the element (asynchronously).

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
I ❤️  the smaller dependency stack. Seems like something is wrong with the tests

**Comment by mkmik** [comment]
yeah EOD snuck upon me before my laptop managed to compile and run the tests so I hoped CI would just pass...

**Comment by mkmik** [comment]
@alamb the tests are passing now

File: server/src/lib.rs [hunk 1]
```diff
@@ -946,27 +981,17 @@ impl ConnectionManager for ConnectionManagerImpl {
         &self,
         connect: &str,
     ) -> Result<Arc<Self::RemoteServer>, ConnectionManagerError> {
-        let ret = cached_remote_server(connect.to_string()).await?;
-        debug!(was_cached=%ret.was_cached, %connect, "getting remote connection");
-        Ok(ret.value)
+        let ret = self
+            .cache
```
Comment (alamb) [inline server/src/lib.rs:985]: 👍

**Review by alamb — APPROVED** [review]
LGTM -- certainly fewer dependencies 👍

**Pull request merged**